src/tests/java/htsjdk/samtools/cram/VersionTest.java

```diff
@@ -0,0 +1,111 @@
+package htsjdk.samtools.cram;
+
+import htsjdk.samtools.CRAMFileWriter;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.cram.build.CramIO;
+import htsjdk.samtools.cram.common.CramVersions;
+import htsjdk.samtools.cram.common.Version;
+import htsjdk.samtools.cram.io.CramInt;
+import htsjdk.samtools.cram.io.InputStreamUtils;
+import htsjdk.samtools.cram.ref.ReferenceSource;
+import htsjdk.samtools.cram.structure.Block;
+import htsjdk.samtools.cram.structure.Container;
+import htsjdk.samtools.cram.structure.ContainerIO;
+import htsjdk.samtools.cram.structure.CramHeader;
+import htsjdk.samtools.seekablestream.SeekableMemoryStream;
+import htsjdk.samtools.seekablestream.SeekableStream;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.zip.CRC32;
+
+/**
+ * Created by vadim on 18/02/2016.
+ */
+public class VersionTest {
+    /**
+     * The test purpose is to ensure that a CRAM written by {@link CRAMFileWriter} adheres to CRAM3 specs expectations:
+     * 1. version 3.+, via both actual byte comparison and CramIO API
+     * 2. EOF container
+     * 3. trailing 4 bytes of a container bytes are a valid crc32 of previous bytes in the container
+     * 3. trailing 4 bytes of a block bytes are a valid crc32 of previous bytes in the block
+     * @throws IOException
+     */
+    @Test
+    public void test_V3() throws IOException {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        ReferenceSource source = new ReferenceSource();
+        SAMFileHeader samFileHeader = new SAMFileHeader();
+        Version version = CramVersions.CRAM_v3;
+        CRAMFileWriter w = new CRAMFileWriter(baos, source, samFileHeader, null);
+        SAMRecord record = new SAMRecord(samFileHeader);
+        record.setReadName("name");
+        record.setAlignmentStart(SAMRecord.NO_ALIGNMENT_START);
+        record.setReferenceIndex(SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX);
+        record.setReadUnmappedFlag(true);
+        record.setReadBases("AAA".getBytes());
+        record.setBaseQualities("!!!".getBytes());
+        w.addAlignment(record);
+        w.close();
+
+        byte[] cramBytes = baos.toByteArray();
+        // expecting 2529 bytes but the size may vary in the future:
+        Assert.assertTrue(cramBytes.length > 100);
+
+        // ensure version bytes:
+        Assert.assertEquals(cramBytes[4], version.major);
+        Assert.assertEquals(cramBytes[5], version.minor);
```

Comment:
I would remove lines 55-60. The tests in the code that follow below seem like the right way to test this, and the 100 byte threshold doesn't seem to have any meaning.